src/set_theory/cardinal_ordinal.lean

```diff
@@ -905,3 +905,17 @@ le_refl _
 end bit
 
 end cardinal
+
+lemma not_injective_of_ordinal {α : Type u} (f : ordinal.{u} → α) :
+  ¬ function.injective f :=
+begin
+  let g : ordinal.{u} → ulift.{u+1} α := λ o, ulift.up (f o),
+  suffices : ¬ function.injective g,
+  { intro hf, exact this (equiv.ulift.symm.injective.comp hf) },
+  intro hg,
+  replace hg := cardinal.mk_le_of_injective hg,
+  rw ← cardinal.lift_mk at hg,
+  have := hg.trans_lt (cardinal.lift_lt_univ _),
+  rw cardinal.univ_id at this,
+  exact lt_irrefl _ this
+end
```

Comment:
Does it make sense to use `small` here, either directly:
```suggestion
lemma not_injective_of_ordinal {α : Type v} [small.{u} α] (f : ordinal.{u} → α) :
  ¬ function.injective f :=
begin
  let g : ordinal.{u} → ulift.{u+1} (shrink α) := (equiv_shrink _).trans equiv.ulift.symm ∘ f,
  suffices : ¬ function.injective g,
  { intro hf, exact this ((equiv.injective _).comp hf) },
  intro hg,
  replace hg := cardinal.mk_le_of_injective hg,
  rw ← cardinal.lift_mk at hg,
  have := hg.trans_lt (cardinal.lift_lt_univ _),
  rw cardinal.univ_id at this,
  exact lt_irrefl _ this
end
```
or via another lemma
```lean
lemma not_injective_of_ordinal' {α : Type v} [small.{u} α] (f : ordinal.{u} → α) :
  ¬ function.injective f :=
begin
  intro hf,
  apply not_injective_of_ordinal (equiv_shrink α ∘ f),
  exact (equiv_shrink _).injective.comp hf,
end
```
The advantage of `small` is that universe unification failure instead turns into a typeclass obligation.

Reply:
Another advantage is that you can then apply this when `α : Type u` and `ordinal.{max u v} → α` thanks to the `small_max` instance.

Reply:
In the use case I imagine, you only need the case where the universe is the same (you're doing some induction in `ordinal.{u}` and you want to show that it stops at some point -- you will choose `u` to be the universe of the type that is relevant for your induction). But I can definitely add the primed version.